lib/api/Healthcheck.js

```diff
@@ -0,0 +1,112 @@
+'use strict'; // eslint-disable-line strict
+
+// In-Sync Replicas
+const ISRS = 3;
+
+/**
+ * Handles healthcheck routes
+ *
+ * @class
+ */
+class Healthcheck {
+    /**
+     * @constructor
+     * @param {object} repConfig - extensions.replication configs
+     * @param {node-zookeeper-client.Client} zkClient - zookeeper client
+     * @param {BackbeatProducer} crrProducer - producer for CRR topic
+     * @param {BackbeatProducer} crrStatusProducer - CRR status producer
+     * @param {BackbeatProducer} metricProducer - producer for metric
+     */
+    constructor(repConfig, zkClient, crrProducer, crrStatusProducer,
+    metricProducer) {
+        this._repConfig = repConfig;
+        this._zkClient = zkClient;
+        this._crrProducer = crrProducer;
+        this._crrStatusProducer = crrStatusProducer;
+        this._metricProducer = metricProducer;
+    }
+
+    _checkProducersReady() {
+        return this._crrProducer.isReady() && this._metricProducer.isReady()
+            && this._crrStatusProducer.isReady();
+    }
+
+    _getConnectionDetails() {
+        return {
+            zookeeper: {
+                status: this._zkClient.getState().name === 'SYNC_CONNECTED' ?
+                    'ok' : 'error',
+                details: this._zkClient.getState(),
+            },
+            kafkaProducer: {
+                status: this._checkProducersReady() ? 'ok' : 'error',
+            },
+        };
+    }
+
+    /**
+     * Checks health of in-sync replicas
+     * @param {object} md - topic metadata object
+     * @return {string} 'ok' if ISR is healthy, else 'error'
+     */
+    _checkISRHealth(md) {
+        // eslint-disable-next-line consistent-return
+        const keys = Object.keys(md);
+        for (let i = 0; i < keys.length; i++) {
+            if (md[keys[i]].isr && md[keys[i]].isr.length !== ISRS) {
+                return 'error';
```

Comment:
Eventually I would prefer a status like "replicas_not_in_sync", more explanatory than "error".

Reply:
I mentioned in the docs that this would return "error" or "ok". There are a few fields like this where the value will be one or the other. I realized that the healthcheck is providing a lot of text, and would prefer to try and narrow it down as much as possible. Hopefully to be figured out during refactors